src/plugins/data/public/search/session/session_service.ts

```diff
@@ -213,6 +213,10 @@ export class SessionService {
    */
   public start() {
     if (!this.currentApp) throw new Error('this.currentApp is missing');
+
+    // cancel previous session if needed
+    this.cleanupPreviousSession();
```

Comment:
not sure I understand the full lifecycle so please correct me - this function calls "cancel" under the hood so is it possible to have a race-condition between creating a new search and the time it takes to cancel the current one so that we cancel the new search?

FWIW, I was not able to reproduce this, purely based on my current understanding.